flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1916 @@
+                        .addEndpoint(FlightEndpoint.newBuilder()
+                                .setTicket(ticket(command))
+                                .setExpirationTime(expirationTime())
+                                .build())
+                        .setTotalRecords(totalRecords())
+                        .setTotalBytes(-1)
+                        .build();
+            }
+
+            @Override
+            public Table resolve() {
+                checkForResolve(command);
+                final Table table = CommandHandlerFixedBase.this.table(command);
+                final long totalRecords = totalRecords();
+                if (totalRecords != -1) {
+                    if (table.isRefreshing()) {
+                        throw new IllegalStateException(
+                                "TicketHandler implementation error; should only override totalRecords for non-refreshing tables");
+                    }
+                    if (table.size() != totalRecords) {
+                        throw new IllegalStateException(
+                                "Ticket handler implementation error; totalRecords does not match the table size");
+                    }
+                }
+                return table;
+            }
+        }
+    }
+
+    private static final class UnsupportedCommand implements CommandHandler, TicketHandler {
+        private final Descriptor descriptor;
+        private final Class<? extends Message> clazz;
+
+        UnsupportedCommand(Descriptor descriptor, Class<? extends Message> clazz) {
+            this.descriptor = Objects.requireNonNull(descriptor);
+            this.clazz = Objects.requireNonNull(clazz);
+        }
+
+        @Override
+        public TicketHandler initialize(Any any) {
+            unpackOrThrow(any, clazz);
+            return this;
+        }
+
+        @Override
+        public boolean isOwner(SessionState session) {
+            return true;
+        }
+
+        @Override
+        public FlightInfo getInfo(FlightDescriptor descriptor) {
+            throw error(Code.UNIMPLEMENTED,
+                    String.format("command '%s' is unimplemented", this.descriptor.getFullName()));
+        }
+
+        @Override
+        public Table resolve() {
+            throw error(Code.INVALID_ARGUMENT, String.format(
+                    "client is misbehaving, should use getInfo for command '%s'", this.descriptor.getFullName()));
+        }
+    }
+
+    // TODO: consider this owning Table instead (SingletonLivenessManager or has + patch from Ryan)
+    abstract class QueryBase implements CommandHandler, TicketHandlerReleasable {
+        private final ByteString handleId;
+        protected final SessionState session;
+
+        private ScheduledFuture<?> watchdog;
+
+        private boolean initialized;
+        private boolean resolved;
+
+        // private final SingletonLivenessManager manager;
+        protected Table table;
```

Comment:
Given your cleanup strategy, I believe this should be private and only accessed via a synchronized accessor that first calls `manageWithCurrentScope` before returning the table. I'm also ready to discuss more about any questions about an alternative liveness approach.